src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java

```diff
@@ -142,12 +149,23 @@ public void identifyWithCompany() {
         verify(intercom).updateUser(any(UserAttributes.class));
     }
 
-
     @Test
-    public void track() {
-        integration.track(new TrackPayloadBuilder().event("Foo").build());
+    public void identifyWithIllegalNestedProperties() {
```

Comment:
Do we also spec passing in an instance of type `Array`? Should we guard against this instead of just `products`

I think if the logic is changes to accept what Intercom accepts, you won't need to worry about nested objects/arrays getting through

Reply:
Cool, just guarded against passing instances of Collection type, which includes Lists and Sets.